main/snappy/APKBUILD

```diff
@@ -1,37 +1,46 @@
 # Contributor: Natanael Copa <ncopa@alpinelinux.org>
 # Maintainer: Natanael Copa <ncopa@alpinelinux.org>
 pkgname=snappy
-pkgver=1.1.4
-pkgrel=3
+pkgver=1.1.7
+pkgrel=0
 pkgdesc="Fast compression and decompression library"
-url="http://google.github.io/snappy/"
+url="https://google.github.io/snappy/"
 arch="all"
 license="BSD-3-Clause"
+makedepends="cmake"
 subpackages="$pkgname-dbg $pkgname-dev $pkgname-doc"
-source="https://github.com/google/snappy/releases/download/$pkgver/snappy-$pkgver.tar.gz
-	fix-UnalignedCopy64.patch
-	"
-[ "$CARCH" = "armhf" ] && options="!check"     # does not pass testsuite on armhf
+source="$pkgname-$pkgver.tar.gz::https://github.com/google/snappy/archive/$pkgver.tar.gz"
 
-builddir="$srcdir/$pkgname-$pkgver"
+[ "$CARCH" = "armhf" ] && options="!check"     # does not pass testsuite on armhf
 
 build() {
 	cd "$builddir"
-	./configure \
-		--prefix=/usr \
-		--disable-static
-	make
+	cmake . \
+		-DCMAKE_INSTALL_PREFIX=/usr \
+		-DCMAKE_INSTALL_LIBDIR=/usr/lib \
+		-DBUILD_SHARED_LIBS=ON
+	cmake --build .
 }
 
 check() {
 	cd "$builddir"
-	make check
+	ctest --verbose
 }
 
 package() {
 	cd "$builddir"
 	make DESTDIR="$pkgdir" install
 }
 
-sha512sums="4f522f1d541e3f00d82ccaf6dda5c6ede28d4db19c53717003d8057261fe8630516ed6ff2a28615da032e0de9c6200b560ed5fce4f8714df9916d50d6a7c2178  snappy-1.1.4.tar.gz
-a086dbdfe2cb21419b33c2f8a2739224a8e843f2c6a51715100c723b9fbd9aed14bcfa791c20dae785b33ac1d1cf4812ed2c5c5d4058c5b8e56a0e004e9a8c60  fix-UnalignedCopy64.patch"
+doc() {
+	default_doc
+	pkgdesc="$pkgdesc documents"
+	mkdir -p "$subpkgdir"/usr/share/"$pkgname"
+	local i;
+	for i in AUTHORS CONTRIBUTING.md COPYING NEWS README.md \
+		format_description.txt framing_format.txt; do
+		cp "$builddir"/$i "$subpkgdir"/usr/share/"$pkgname"/
+	done
```

Comment:
Documentation files should be installed into `/usr/share/doc/$pkgname`.

Please move the files to the desired location inside `package()` function and remove custom `doc()` function.